aten/src/ATen/core/boxing/KernelFunction.h

```diff
@@ -0,0 +1,466 @@
+#pragma once
+
+#include <ATen/core/stack.h>
+#include <c10/util/TypeList.h>
+#include <ATen/core/boxing/kernel_functor.h>
+#include <ATen/core/boxing/kernel_function.h>
+#include <ATen/core/boxing/kernel_lambda.h>
+
+namespace c10 {
+
+namespace detail {
+template<class Type>
+constexpr uint64_t hashType() {
+  return typeid(Type).hash_code()
```

Comment:
Doesn't this imply RTTI?

Reply:
I guess, because this is constexpr, technically it's not? But I am not so sure about the implications.

Reply:
I'm not 100% sure either. I'm trusting CI here. If this turns out to be impossible, it's easy to disable - it's just for error checking.

Reply:
did you verify (diasm in gdb or something) that it indeed becomes a constexpr value?

Reply:
shit it actually does not. Good call. I could use `CAFFE_KNOWN_TYPE`, but that wouldn't be constexpr either. I'll remove the error checking code for now. I have an idea of how to do with constexpr it in C++14, maybe I can port it to C++11, but that'll be another diff.

Reply:
Update: My idea works with C++11, but only GCC >= 5. So seems we need to wait for that to happen.